### PR TITLE
[Functorch] Delete unused files

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83464
* __->__ #83777
* #84788

Differential Revision: [D39478381](https://our.internmc.facebook.com/intern/diff/D39478381)

@diff-train-skip-merge Manually copied this commit to fbsync to fix shipit

### CODE REVIEW

File: functorch/.github/workflows/docs.yml [hunk 1]
```diff
@@ -1,82 +0,0 @@
-name: Build and Deploy Docs
-on:
-  pull_request:
-    types: [opened, synchronize, reopened]
-  push:
-    branches:
-      - main
```
Comment (zou3519) [inline functorch/.github/workflows/docs.yml:7]: We still need the docs workflow; we have not migrated it over to PyTorch yet

Comment (zou3519) [reply functorch/.github/workflows/docs.yml:7]: Actually, let me think about this for a little more (it depends on what the next PR up in the stack is doing)

Comment (malfet) [reply functorch/.github/workflows/docs.yml:7]: Well, I don't think it works that way, is it? (Unless this folder is still mirrored to github.com/pytorch/functorch)

Comment (zou3519) [reply functorch/.github/workflows/docs.yml:7]: >  (Unless this folder is still mirrored to github.com/pytorch/functorch)

It is, for now at least

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39032967).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39032967).

**Review by zou3519 — APPROVED** [review]
LGTM, but please wait until we disable shipit for pytorch/functorch to merge this

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39058811).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39032967).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D39478381).

**Pull request closed without merging**